File: mobile/lib/core/models/auth_result.dart
class AuthResult {
  AuthResult({
    required this.accessToken,
    required this.refreshToken,
    required this.isNewUser,
    this.primaryPhoneHint,
    this.primaryEmailHint,
  });

  final String accessToken;
  final String refreshToken;
  final bool isNewUser;
  final String? primaryPhoneHint;
  final String? primaryEmailHint;

  factory AuthResult.fromJson(Map<String, dynamic> json) {
    return AuthResult(
      accessToken: json['access_token']?.toString() ?? '',
      refreshToken: json['refresh_token']?.toString() ?? '',
      isNewUser: json['is_new_user'] == true,
      primaryPhoneHint: json['primary_phone_hint']?.toString(),
      primaryEmailHint: json['primary_email_hint']?.toString(),
    );
  }
}

File: backend/services/auth-service/internal/domain/model/auth_result.go
package model

import "time"

type AuthResult struct {
	AccessToken      string    `json:"access_token"`
	RefreshToken     string    `json:"refresh_token"`
	TokenType        string    `json:"token_type,omitempty"`
	ExpiresAt        time.Time `json:"expires_at,omitempty"`
	RefreshExpiresAt time.Time `json:"refresh_expires_at,omitempty"`
	SessionID        string    `json:"session_id,omitempty"`
	IsNewUser        bool      `json:"is_new_user"`
	PrimaryPhoneHint *string   `json:"primary_phone_hint,omitempty"`
	PrimaryEmailHint *string   `json:"primary_email_hint,omitempty"`
}

// DeviceInfo describes the client device. Mirrors token-service's DeviceInfo
// at the auth-service boundary so auth-service stays decoupled from tokenpb types.
type DeviceInfo struct {
	DeviceID   string
	Platform   string
	OSVersion  string
	AppVersion string
	Model      string
	UserAgent  string
	IPAddress  string
}
